avro-fastserde/src/test/java/com/linkedin/avro/fastserde/FastStringableTest.java

```diff
@@ -132,6 +136,30 @@ public void serializeStringableFields() throws URISyntaxException, MalformedURLE
     }
   }
 
+  @Test(groups = {"deserializationTest"}, dataProvider = "SlowFastDeserializer")
+  public void javaStringPropertyTest(Boolean whetherUseFastDeserializer) throws IOException {
+    Schema javaStringSchema = Schema.parse("{\n" + "  \"type\":\"string\",\n" + "  \"avro.java.string\":\"String\"} ");
+    Schema schema = createRecord(createField("testString", javaStringSchema));
+
+    GenericRecord record = new GenericData.Record(schema);
+    record.put("testString", "aaa");
+
+    Decoder decoder = writeWithFastAvro(record, schema, false);
+
+    GenericRecord afterDecoding;
+    if (whetherUseFastDeserializer) {
+      afterDecoding = readWithFastAvro(schema, schema, decoder, false);
+    } else {
+      afterDecoding = readWithSlowAvro(schema, schema, decoder, false);
+    }
+
+    if (Utils.isAvro14()){
+      Assert.assertTrue(afterDecoding.get(0) instanceof Utf8, "Utf8 is expected, but got: " + afterDecoding.get(0).getClass());
+    }  else {
+      Assert.assertTrue(afterDecoding.get(0) instanceof String, "String is expected, but got: " + afterDecoding.get(0).getClass());
```

Comment:
I may miss some context here, so `String` would be the default format after Avro 1.4?

Reply:
Avro-1.4 doesn't recognize the modern hint  defined in the schema: `avro.java.string`, so the de-serialized result will  be Utf8, and those hints were introduced in avro-1.7, and Avro will convert the string field to Java String.
GenericDatumReader.java in avro-1.7:
```
protected Class findStringClass(Schema schema) {
    String name = schema.getProp(GenericData.STRING_PROP);
    if (name == null) return CharSequence.class;

    switch (GenericData.StringType.valueOf(name)) {
      case String:
        return String.class;
      default:
        return CharSequence.class;
    }
  }
```